demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -1408,7 +1385,7 @@ int main( int argc,
              * attempts are reached or maximum timeout value is reached. The function
              * returns EXIT_FAILURE if the TCP connection cannot be established to
              * broker after configured number of attempts. */
-            returnStatus = connectToServerWithBackoffRetries( &networkContext );
+            returnStatus = connectToServerWithBackoffRetries( &networkContext, &mqttContext, &clientSessionPresent, &brokerSessionPresent );
```

Comment:
Why is the `clientSessionPresent` passed as an output parameter? Shouldn't it just be an input parameter?
It would be more readable if the value of `clientSessionPresent` variable is set after the call to `connectToServerWithBackOffRetries`
```
            returnStatus = connectToServerWithBackoffRetries( &networkContext, &mqttContext, &clientSessionPresent, &brokerSessionPresent );
       
            if( returnStatus == EXIT_FAILURE )
            {
                /* Log error to indicate connection failure after all
                 * reconnect attempts are over. */
                LogError( ( "Failed to connect to MQTT broker %.*s.",
                            BROKER_ENDPOINT_LENGTH,
                            BROKER_ENDPOINT ) );
            }
            else
            { 
       
              /* Update the flag to indicate that an MQTT client session is saved.
               * Once this flag is set, MQTT connect in the following iterations of
                * this demo will be attempted without requesting for a clean session. */ 
               clientSessionPresent = true;     
```